controllers/localVolumeCleanUp.go

```diff
@@ -1,76 +1,119 @@
 package controllers
 
-// import (
-// 	"context"
-// 	"fmt"
-// 	"strings"
+import (
+	"context"
+	"fmt"
+	"strings"
 
-// 	corev1 "k8s.io/api/core/v1"
-// 	"k8s.io/apimachinery/pkg/api/errors"
-// )
+	localv1 "github.com/openshift/local-storage-operator/pkg/apis/local/v1"
+	corev1 "k8s.io/api/core/v1"
+	apiextenstionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
 
-// // fetchRemovePV finds and deletes all local volume PVs
-// func (r *CleanUpWatcher) fetchRemovePV(ctx context.Context) bool {
-// 	// Find PVs
-// 	persistenceVolume := []corev1.PersistentVolume{}
-// 	pvList := &corev1.PersistentVolumeList{}
-// 	err := r.List(ctx, pvList)
-// 	if err != nil {
-// 		fmt.Print("Error in Getting PV List")
-// 		return false
-// 	}
-// 	for _, pv := range pvList.Items {
-// 		if strings.HasPrefix(pv.Name, "local-pv-") {
-// 			persistenceVolume = append(persistenceVolume, pv)
-// 			fmt.Println("PV status- ", pv.Status.Phase)
-// 		}
-// 	}
-// 	// PV Deletion
-// 	for _, pv := range persistenceVolume {
-// 		err = r.Delete(ctx, &pv)
-// 		if err != nil && !errors.IsNotFound(err) {
-// 			fmt.Print("Error in Deleting PV ", pv.Name)
-// 			return false
-// 		}
-// 	}
-// 	fmt.Println("PV Deleted.....")
-// 	return true
-// }
+func (cr *CleanUpOperatorReconciler) localVolumeCleanUp(ctx context.Context, namespace string) error {
+	localVolumeStatus := true
+	localDisk := &localv1.LocalVolume{}
+	err := cr.Get(ctx, types.NamespacedName{Name: "local-disk", Namespace: namespace}, localDisk)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			fmt.Println("LocalVolume 'local-disk' not found")
+			localVolumeStatus = false
+		} else {
+			fmt.Println(err, "Error in getting LocalVolume 'local-disk'")
+			return err
+		}
+	}
 
-// // deleteMountedPath deletes mounted path from each node
-// func (r *CleanUpWatcher) deleteMountedPath(ctx context.Context) bool {
-// 	// Remove Mounted Path
-// 	nodesList := &corev1.NodeList{}
-// 	err := r.List(ctx, nodesList)
-// 	if err != nil {
-// 		fmt.Print("Error in Getting Nodes List")
-// 		return false
-// 	}
+	if localVolumeStatus {
+		localDisk.SetFinalizers([]string{})
+		if err := cr.Update(ctx, localDisk); err != nil {
+			fmt.Println(err, "Error is removing finalizers from Local-Volume 'local-disk'")
+			return err
+		}
+	}
 
-// 	for _, node := range nodesList.Items {
-// 		command := "oc debug node/" + node.Name + " -- chroot /host rm -rf /mnt"
-// 		_, out, _ := ExecuteCommand(command)
-// 		fmt.Println(out)
-// 	}
-// 	fmt.Println("Mounted Paths Removed....")
-// 	return true
-// }
+	// Find PVs
+	pvList := &corev1.PersistentVolumeList{}
+	pvfound := true
+	err = cr.List(ctx, pvList)
+	if err != nil {
+		fmt.Println(err, "Error in getting PVs")
+		pvfound = false
```

Comment:
I think we need to return err if there is any error while getting PV list.
I asked just not to remove IsNotFound error